addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1198,7 +1198,7 @@ const Wysiwyg = Widget.extend({
         const backgroundImage = $(targetElement).css('background-image');
         let backgroundGradient = false;
         if (weUtils.isColorGradient(backgroundImage)) {
-            const textGradient = targetElement.classList.contains('text-gradient');
+            const textGradient = $(targetElement).css('-webkit-background-clip') === 'text';
```

Comment:
That needs comments in the code, otherwise it's really impossible to know why you are not checking the text-gradient class